libclamav_rust/src/fmap.rs

```diff
@@ -0,0 +1,102 @@
+/*
+ *  Rust interface for libclamav FMap module
+ *
+ *  Copyright (C) 2023 Cisco Systems, Inc. and/or its affiliates. All rights reserved.
+ *
+ *  Authors: Micah Snyder
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License version 2 as
+ *  published by the Free Software Foundation.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ *  MA 02110-1301, USA.
+ */
+
+use std::convert::TryFrom;
+
+use log::{debug, error};
+use thiserror::Error;
+
+use crate::{sys, util::optional_cstring};
+
+/// FMapError enumerates all possible errors returned by this library.
+#[derive(Error, Debug)]
+pub enum FMapError {
+    #[error("Invalid parameter: {0}")]
+    InvalidParameter(String),
+
+    #[error("{0} parmeter is NULL")]
+    NullParam(&'static str),
+
+    #[error("Offset {0} and length {1} not contained in FMap of size {2}")]
+    NotContained(usize, usize, usize),
+
+    #[error("FMap pointer not initialized: {0}")]
+    UninitializedPtr(&'static str),
+}
+
+#[derive(PartialEq, Eq, Hash, Debug)]
+pub struct FMap {
+    fmap_ptr: *mut sys::cl_fmap_t,
+}
```

Comment:
If this is all this struct will ever contain, it's probably going to save you a few keystrokes to just make it a tuple struct:

```suggestion
pub struct FMap(*mut sys::cl_fmap_t);
```

Reply:
At least for now.  Someday I imagine we will want to change it from a wrapper to actually implement the fmap stuff in Rust. 

So TBH not sure if a tuple is better or worse for the future.